src/stateAggregator/accessors/orgAccessor.ts

```diff
@@ -0,0 +1,187 @@
+
+  public getAll(decrypt = false): SfOrgs {
+    return this.globalInfo.get(SfInfoKeys.ORGS, decrypt);
+  }
+
+  public get(username: string, decrypt = false): Optional<SfOrg> {
+    const auth = this.globalInfo.get<SfOrg>(`${SfInfoKeys.ORGS}["${username}"]`, decrypt);
+    // For legacy, some things wants the username in the returned auth info.
+    if (auth && !auth.username) auth.username = username;
+    return auth;
+  }
+
+  public has(username: string): boolean {
+    return !!this.getAll()[username];
+  }
+
+  public set(username: string, org: SfOrg): void {
+    // For legacy, and to keep things standard, some things wants the username in auth info.
+    org.username ??= username;
+    this.globalInfo.set(`${SfInfoKeys.ORGS}["${username}"]`, org);
+  }
+
+  public update(username: string, org: Partial<SfOrg>): void {
+    // For legacy, and to keep things standard, some things wants the username in auth info.
+    org.username ??= username;
+    this.globalInfo.update(`${SfInfoKeys.ORGS}["${username}"]`, org);
+  }
+
+  public unset(username: string): void {
+    delete this.globalInfo.get(SfInfoKeys.ORGS)[username];
+  }
+}
+
+export abstract class BaseOrgAccessor<T extends ConfigFile, P extends ConfigContents> {
+  private configs: Map<string, Nullable<T>> = new Map();
+  private contents: Map<string, P> = new Map();
+  private files: string[];
+
+  public constructor() {
+    this.files = this.getAllFiles();
+  }
+
+  public async read(username: string, decrypt = false, throwOnNotFound = true): Promise<P> {
+    try {
+      const config = await this.initAuthFile(username, throwOnNotFound);
+      this.configs.set(username, config);
+      return this.get(username, decrypt) as P;
+    } catch (err) {
+      return {} as P;
+    }
+  }
+
+  public async readAll(decrypt = false): Promise<P[]> {
+    for (const file of this.files) {
+      const username = this.parseUsername(file);
+      const config = await this.initAuthFile(username);
+      this.configs.set(username, config);
+    }
+    return this.getAll(decrypt);
+  }
+
+  public get(username: string, decrypt = false): Nullable<P> {
+    const config = this.configs.get(username);
+    if (config) {
+      this.contents.set(username, config.getContents(decrypt) as P);
+    }
+
+    return this.contents.get(username);
+  }
+
+  public getAll(decrypt = false): P[] {
+    return [...this.configs.keys()].reduce((orgs, username) => {
+      const org = this.get(username, decrypt);
+      return org && !isEmpty(org) ? orgs.concat([org]) : orgs;
+    }, [] as P[]);
+  }
+
+  public has(username: string): boolean {
+    return this.contents.has(username);
+  }
+
+  public async exists(username: string): Promise<boolean> {
+    const config = this.configs.get(username);
+    return config ? await config.exists() : false;
+  }
+
+  public hasFile(username: string): boolean {
+    return Boolean(this.files.find((f) => this.parseUsername(f) === username));
```

Comment:
`some` is already boolean
`return this.files.some((f) => this.parseUsername(f) === username))`